source/_integrations/homekit_controller.markdown

```diff
@@ -45,66 +45,77 @@ ha_integration_type: integration
-- Light (HomeKit lights)
-- Lock (HomeKit lock)
-- Switch (HomeKit switches, outlets and valves)
-- Binary Sensor (HomeKit motion, contact, occupancy, carbon monoxide and smoke sensors)
-- Sensor (HomeKit humidity, temperature, co2 and light level sensors)
-- Fan
-- Air Quality
-- Humidifier (HomeKit humidifiers and dehumidifiers)
-- Automation Triggers (HomeKit 'stateless' accessories like buttons, remotes and doorbells)
-
-<div class='note'>
+- [via Ethernet or Wi-Fi](#adding-a-homekit-device-via-ethernet-or-wi-fi)
+- [via Bluetooth](#adding-a-homekit-device-through-bluetooth)
+- via Thread
+  - [by using Home Assistant’s preferred Thread network](#adding-a-homekit-device-to-a-thread-network-via-home-assistant)
+  - [by using Apple Thread border router](#adding-a-homekit-device-via-apple-thread-border-router)
 
-If your device is not on this list then you may still be able to pair it, and it will be visible in the Device Registry. But Home Assistant may not create entities for it.
+## Adding a HomeKit device via Ethernet or Wi-Fi
 
-</div>
+The HomeKit controller integration automatically detects HomeKit [compatible devices](#supported-devices) on your network when they are ready to pair.
 
-HomeKit IP accessories for these device types may work with some caveats:
-
-- If the device is Wi-Fi based and has no physical controls or screen then you may need an Apple HomeKit device like an iPhone or iPad to get the accessory onto your Wi-Fi network. For example, for a Koogeek LS1 you must add the accessory to HomeKit on your iOS device, then remove it from the iOS device. This leaves the LS1 in an unpaired state but still on your Wi-Fi. Home Assistant can then find it and pair with it.
-- You need to know the HomeKit PIN. There is no way to recover this if you do not have it. In this case, you will need to contact the manufacturer to see what options you have.
+### Prerequisites
 
-The integration will poll your devices, but it will also automatically enable push updates for accessories that support it.
+- If you do not have the [`default_config`](/integrations/default_config/) integration, add [`zeroconf`](/integrations/zeroconf/) to your `configuration.yaml` file.
+- Find your HomeKit pairing code. The code is on the device itself, or on the packaging. If your device has a screen, it may be shown on screen.
+   - There is no way to recover this if you do not have it. In this case, you will need to contact the manufacturer to see what options you have.
+- Make sure your device is powered up.
+- Make sure the device is on your network, but not paired with another HomeKit controller. Depending on the device, you need to follow a different set of steps:
+  - If your device is not already in your network: Join the device to your network:
+    - Follow the instructions of the manufacturer on how to join the device to the network
+    - If the device is HomeKit only, pair it with the Apple Home app, then follow the next step.
```

Comment:
```suggestion
    - If the device is HomeKit only, pair it with the Apple Home app, then follow the next step (e.g. Koogeek LS1 Light Strip).
```

Reply:
```suggestion
    - If the device is HomeKit only (for example the Koogeek LS1 light strip), pair it with the Apple Home app, then follow the next step.
```
otherwise it sounds like there is a step specific to the light strip.